daemon/src/org/eclipse/iofog/tracking/TrackingEventsStorage.java

```diff
@@ -0,0 +1,30 @@
+package org.eclipse.iofog.tracking;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class TrackingEventsStorage {
+    private static TrackingEventsStorage instance = null;
+    public static TrackingEventsStorage getInstance() {
+        if (instance == null) {
+            synchronized (TrackingEventsStorage.class) {
+                if (instance == null)
+                    instance = new TrackingEventsStorage();
+            }
+        }
+        return instance;
```

Comment:
You use TrackingEventsStorage only inside Tracker. I think you don't need singleton here. Just create reference to it in Tracker.